### ISSUE TITLE
Generating openapi from proto gives field of type string and format RFC3339 but not vice versa

### ISSUE THREAD

**Issue by panperla** [google/gnostic#310]
I need such output in my proto 
```
...
message XYZ {
    string versionName = 1 [json_name = "versionname"];
    string branchName = 2 [json_name = "branchname"];
    string appName = 3 [json_name = "appname"];
    string appDomainName = 4 [json_name = "appdomainname"];
    string appURL = 5 [json_name = "appurl"];
    string commitSHA = 6 [json_name = "commitsha"];
    google.protobuf.Timestamp responseTime = 7 [json_name = "responsetime"];
}
...
```
During the conversion from openapi google.protobuf.Timestamp type is being just translated as string.
In openapi I'm defining it as a:
```
responsetime:
                    type: string
                    format: RFC3339
```

To conclude generating openapi file from proto gives me field with type string and format RFC3339 but when I convert it back to proto the google.protobufTImestamp is lost.

I might be missing something please point me into right direction how I can arrive with  google.protobuf.Timestamp responseTime from openapi.

**Comment by adolfo** [comment]
I'm having a similar issue, but under a slightly different use case. I'm going from Protobuf to OpenAPI to a Rest Client. In that process I'm losing the timestamp type information and ending up with a plain string as well.

I believe this is caused by an invalid string `format` in the generated OpenAPI spec. Specifically, the timestamp format "RFC3339" isn't a valid format per the OpenAPI docs. It states that the "date-time" format already conforms to RFC 3339:
> OpenAPI defines the following built-in string formats: [...] **date-time** – the date-time notation as defined by RFC 3339, section 5.6, for example, 2017-07-21T17:32:28Z [source1](https://swagger.io/docs/specification/data-models/data-types/) [source2](https://github.com/google/gnostic/blob/ed308846fc77db7667936d27d7ac1fb278b74216/openapiv3/schema-generator/3.0.2.md#format)

It appears gnostic is inserting "RFC3339" where "date-time" appears to be the correct format value:
https://github.com/google/gnostic/blob/3334dd941843bab02ece98e48fd99349e3d88d1f/cmd/protoc-gen-openapi/generator/openapi-v3.go#L766-L770

I propose that the "RFC3339" format be replaced with the correct "date-time" format. This will enable other tools such as [OpenAPI-Generator](https://github.com/OpenAPITools/openapi-generator), etc. to do the right thing when it comes to generating timestamps for each respective language. 

It's not clear if this was an intentional value that is used downstream, or a simple oversight, but I'd gladly open a PR if the maintainers agree with this change. /cc @morphar @timburks

**Comment by adolfo** [comment]
Addressed in #319
